File: CineVibe/CineVibe.Services/Database/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CineVibe.Services.Database
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public decimal Price { get; set; }
        
        public byte[]? Picture { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: CineVibe/UI/cinevibe_mobile/lib/model/product.dart
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  final int id;
  final String name;
  final double price;
  final String? picture; // Base64 encoded image
  final bool isActive;
  final DateTime createdAt;

  Product({
    this.id = 0,
    this.name = '',
    this.price = 0.0,
    this.picture,
    this.isActive = true,
    required this.createdAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}
